Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -384,16 +496,21 @@ private extension Substring {
 
 extension HTTPHeaders: Equatable {
     public static func ==(lhs: HTTPHeaders, rhs: HTTPHeaders) -> Bool {
-        if lhs.storage.count != rhs.storage.count {
+        guard lhs.names.count == rhs.headers.count else {
```

Comment:
Why are these not comparing against the same property?

Reply:
brainfart...